README.md
# tiktactoe

We all have played tiktactoe during our school time, so in college it was time to bring it on a virtual platform.
It is a simple java program using awt for gui purpose.

## Running it locally
1. Clone it locally
2. Open root location on cmd
3. Run "java tiktactoe.index"
4. Game starts running

## User guide
On starting screen you will get 5 buttons:
1. New game - to start a new game
2. Resume - to continue an incomplete game
3. History - to check results of last 10 games
4. About - to know more about game 
5. Exit - to exit the game

## Snapshots
<!-- ![Capture](https://user-images.githubusercontent.com/61378660/149667962-0ba73887-57db-4873-b696-71d84fc017f5.PNG)
![Capture2](https://user-images.githubusercontent.com/61378660/149668119-726271e8-92a2-4e7c-83ee-682fce9eea9c.PNG)
![Capture3](https://user-images.githubusercontent.com/61378660/149668206-df5e6807-e67e-4feb-b63d-5942c7709509.PNG)
![Capture4](https://user-images.githubusercontent.com/61378660/149668214-17050b1a-2132-4d8c-90a1-84bdf98bd158.PNG) -->

<p width="100%">
<img width="24%" src="https://user-images.githubusercontent.com/61378660/149667962-0ba73887-57db-4873-b696-71d84fc017f5.PNG">
<img width="24%" src="https://user-images.githubusercontent.com/61378660/149668119-726271e8-92a2-4e7c-83ee-682fce9eea9c.PNG">
<img width="24%" src="https://user-images.githubusercontent.com/61378660/149668206-df5e6807-e67e-4feb-b63d-5942c7709509.PNG">
<img width="24%" src="https://user-images.githubusercontent.com/61378660/149668214-17050b1a-2132-4d8c-90a1-84bdf98bd158.PNG">
</p>  

## Contributing
Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.

Please make sure to update tests as appropriate.

## License
[MIT](https://choosealicense.com/licenses/mit/)


tiktactoe/History.txt
1
2
1
0
1
1
1
1
1
0


tiktactoe/demo.java
package tiktactoe;
import java.awt.*;
import java.awt.image.*;
import java.io.*;
import javax.imageio.*;
import java.awt.event.*;
public class demo extends Frame implements ActionListener
{
   Image img;
   Button n_game,resume,h_score,about,exit; 
   demo(Image r)
   {
    img=r;
     n_game=new Button("NEW GAME");
     resume=new Button("RESUME");
     h_score=new Button("HISTORY");
     about=new Button("ABOUT");
     exit=new Button("EXIT");
    setUndecorated(true);
    setVisible(true);
    setSize(400,600);
    setLocation(400,100);
    setLayout(null);
    
    setForeground(Color.WHITE);
    setFont(new Font("Berlin Sans FB",Font.BOLD,35));
    

    add(n_game);
    n_game.setSize(250,50);
    n_game.setLocation(75,150);
    n_game.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
    n_game.setForeground(new Color(66,43,15));
    n_game.addActionListener(this);
  
    add(resume);
    resume.setSize(250,50);
    resume.setLocation(75,240);
    resume.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
    resume.setForeground(new Color(66,43,15));
    resume.addActionListener(this);

    add(h_score);
    h_score.setSize(250,50);
    h_score.setLocation(75,330);
    h_score.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
    h_score.setForeground(new Color(66,43,15));
    h_score.addActionListener(this);

    add(about);
    about.setSize(250,50);
    about.setLocation(75,420);
    about.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
    about.setForeground(new Color(66,43,15));
    about.addActionListener(this);

    add(exit);
    exit.setSize(250,50);
    exit.setLocation(75,510);
    exit.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
    exit.setForeground(new Color(66,43,15));
    exit.addActionListener(this);
   }
    
    
   public void actionPerformed(ActionEvent e)
  {
       if(e.getSource()==exit)
       System.exit(0);
       if(e.getSource()==about)
       {about a=new about(img);
        setVisible(false);
       }
       if(e.getSource()==n_game)
       {game g=new game(img,false);
        setVisible(false);
       }
       if(e.getSource()==resume)
       {game g=new game(img,true);
         setVisible(false);
        }
       if(e.getSource()==h_score)
      {
         history h=new history(img);
         setVisible(false);
      }
  }
   public void paint(Graphics g)
  {
     
     g.drawImage(img,0,0,null);
     g.drawString("TIK-TAC-TOE",90,100);
  }
  }

tiktactoe/about.java
package tiktactoe;
import javax.imageio.*;
import java.awt.image.*;
import java.awt.event.*;
import java.awt.*;
import java.io.*;
import java.util.*;
public class about extends Frame implements ActionListener
{
  Button back;
  Image b;
  about(Image a)
  {
    b=a;
    setUndecorated(true);
    setVisible(true);
    setSize(400,600);
    setLocation(400,100);
    setLayout(null);
    setForeground(new Color(66,43,15));
    setFont(new Font("Berlin Sans FB",Font.BOLD,15));
  
    back=new Button("BACK");
    add(back);
    back.setSize(60,30);
    back.setLocation(20,550);
    back.addActionListener(this);
  }
  public void actionPerformed(ActionEvent e)
  {
    if(e.getSource()==back)
    {
       demo d=new demo(b);
       setVisible(false);
    } 
  }
  
  public void paint(Graphics g)
  {
     g.drawImage(b,0,0,null); 
     g.setFont(new Font("Berlin Sans FB",Font.BOLD,35));
     g.setColor(Color.WHITE);
     g.drawString("ABOUT",135,80);
     g.fillRoundRect(0,150,400,380,40,40);
    g.setColor(new Color(66,43,15)); 
     g.setFont(new Font("Constantia",Font.PLAIN,20));
    g.drawString("This game of TIK-TAC-TOE is developed",20,200);
    g.drawString("by Developer Dharmik Bhadra from YBPP.",20,230);
    g.drawString("It is developed as a micro-project for java",20,290);
    g.drawString("under guidance of Mrs Sukeshini Gawai",20,320);
    g.drawString("This game is developed using java awt and",20,350);
    g.drawString("javax.It is a simple TIK-TAC-TOE game.",20,380);
    g.drawString("This game is under CopyRight of BRO ",20,410);
    g.drawString("PVT LTD and Illegal use of game without",20,440);
    g.drawString("the permission is serious criminal offense",20,470);
   g.drawString("under YBPC Law 240,340 and 440.",20,500);
   
  }
public static void main(String args[])
{
  //about a=new about();
}
}

tiktactoe/history.java
package tiktactoe;
import javax.imageio.*;
import java.awt.image.*;
import java.awt.event.*;
import java.awt.*;
import java.io.*;
import java.util.*;
public class history extends Frame implements ActionListener
{
   Image img;
   Button b;
   File f;
   BufferedReader br;
   BufferedWriter bw;
   Vector<String> vec;
   String s;
   int count;
   history(Image rimg)
   {
      count=0;
      img=rimg;
    try{f=new File("D:\\desktop imp\\tiktactoe\\History.txt");
    if(!f.exists())
    f.createNewFile();
    br=new BufferedReader(new FileReader(f));
    }catch(Exception ee){}
    vec=new Vector<String>();
    setSize(400,600);
    setUndecorated(true);
    setVisible(true);
    setSize(400,600);
    setLocation(400,100);
    setLayout(null);
    setForeground(new Color(66,43,15));
    setFont(new Font("Berlin Sans FB",Font.BOLD,30));

    b=new Button("BACK");
    b.setSize(70,20);
    b.setLocation(10,570);
    b.setFont(new Font("Berlin Sans FB",Font.BOLD,20));
    b.addActionListener(this);
    add(b);

    try{
      vec.clear();
       while((s=br.readLine())!=null)
       {   vec.add(s);
           count++; 
       }
     
     }catch(IOException ee){ee.printStackTrace();}
   }
   
   public void actionPerformed(ActionEvent e)
  {
    demo d=new demo(img);
    setVisible(false);
  }
   public void paint(Graphics g)
  {
     int y=55,x2=30,y2=100;
     
     g.drawImage(img,0,0,null);
     g.setColor(Color.WHITE);
     g.drawString("HISTORY",140,35);
     g.fillRoundRect(5,55,390,500,15,15);
     g.setColor(new Color(66,43,15));
     for(int i=0;i<10;i++)
     {
         g.drawLine(5,y,395,y);
         y+=50;
      }
     g.setFont(new Font("Constantia",Font.PLAIN,25));
     if(count<=10)
    {
      for(int i=vec.size()-1;i>=0;i--)
       {
           s=vec.get(i);
          if(!s.equals("0"))
          g.drawString("Player "+s+" won",x2,y2);
          else
          g.drawString("Its an draw",x2,y2); 
          y2+=50;
      }
    }
    else
    {   y2=550;
       try{
       bw=new BufferedWriter(new FileWriter(f));
       for(int i=vec.size()-10;i<vec.size();i++)
       {
           s=vec.get(i);
           bw.write(s);
           bw.newLine();
           if(!s.equals("0"))
          g.drawString("Player "+s+" won",x2,y2);
          else
          g.drawString("Its an draw",x2,y2); 
          y2-=50;
       }
       bw.close();
       }catch(Exception ee){ee.printStackTrace();}
    

    } 
  }
 
}

tiktactoe/game.java
package tiktactoe;
import javax.imageio.*;
import java.awt.image.*;
import java.awt.event.*;
import java.awt.*;
import java.io.*;
import java.util.*;
class game extends Frame implements ActionListener
{  
   Image a;
   Button b1,b2,b3,b4,b5,b6,b7,b8,b9,back;
   Label l;
   Button bb2;
   File f,f2;
   Dialog d;
   BufferedWriter bw,bw2;
   BufferedReader br,br2;
   int i=1;
   int in;
   String s1,s2;
   game(Image r,boolean b)
   {
    
    setSize(400,600);
    setUndecorated(true);
    setVisible(true);
    setSize(400,600);
    setLocation(400,100);
    setLayout(null);
    setForeground(new Color(66,43,15));
    setFont(new Font("Berlin Sans FB",Font.BOLD,40));

    a=r;
    back=new Button("BACK");
    back.setSize(70,20);
    back.setLocation(10,570);
    back.addActionListener(this);
    back.setFont(new Font("Berlin Sans FB",Font.BOLD,20));
    add(back);

    b1=new Button();
    b1.setSize(130,160);
    b1.setLocation(2,70);
    b1.addActionListener(this);
    add(b1);

    b2=new Button();
    b2.setSize(130,160);
    b2.setLocation(135,70);
    b2.addActionListener(this);
    add(b2);

    b3=new Button();
    b3.setSize(130,160);
    b3.setLocation(268,70);
    b3.addActionListener(this);
    add(b3);
    
    b4=new Button();
    b4.setSize(130,160);
    b4.setLocation(2,235);
    b4.addActionListener(this);
    add(b4);

    b5=new Button();
    b5.setSize(130,160);
    b5.setLocation(135,235);
    b5.addActionListener(this);
    add(b5);

    b6=new Button();
    b6.setSize(130,160);
    b6.setLocation(268,235);
    b6.addActionListener(this);
    add(b6);

    b7=new Button();
    b7.setSize(130,160);
    b7.setLocation(2,400);
    b7.addActionListener(this);
    add(b7);

    b8=new Button();
    b8.setSize(130,160);
    b8.setLocation(135,400);
    b8.addActionListener(this);
    add(b8);

    b9=new Button();
    b9.setSize(130,160);
    b9.setLocation(268,400);
    b9.addActionListener(this);
    add(b9);
    
    
    try{
    f=new File("D:\\desktop imp\\tiktactoe\\ScoreBook.txt");
    if(!f.exists())
    f.createNewFile();
    
    f2=new File("D:\\desktop imp\\tiktactoe\\History.txt");
    if(!f2.exists())
    f2.createNewFile();
    
    bw2=new BufferedWriter(new FileWriter(f2,true));
    br=new BufferedReader(new FileReader(f));
    
    if(b)
    {
       while((s1=br.readLine())!=null)
       {
          s2=br.readLine();
          in=Integer.parseInt(s1);
          switch(in)
          {
            case 1:
            b1.setLabel(s2);
            break;
            case 2:
            b2.setLabel(s2);
            break;
            case 3:
            b3.setLabel(s2);
            break;
            case 4:
            b4.setLabel(s2);
            break;
            case 5:
            b5.setLabel(s2);
            break;
            case 6:
            b6.setLabel(s2);
            break;
            case 7:
            b7.setLabel(s2);
            break;
            case 8:
            b8.setLabel(s2);
            break;
            case 9:
            b9.setLabel(s2);
            break;
         }
       }
    }
    bw=new BufferedWriter(new FileWriter(f));
    }catch(Exception ee){}
   }
   public void actionPerformed(ActionEvent e)
  {
      Button b=(Button)e.getSource();
      String s=b.getLabel();
      
      if(s.equals(""))
     {
      if(i%2==1)
      b.setLabel("X");
      else
      b.setLabel("O");
      i++;
     }
     if(e.getSource()==back)
     {
        demo dd=new demo(a);
        
        setVisible(false);
        try{
        if(!(b1.getLabel().equals("")))
        { 
           bw.write("1");
           bw.newLine();
           bw.write(b1.getLabel());
           bw.newLine();
        }
        if(!(b2.getLabel().equals("")))
        { 
           bw.write("2");
           bw.newLine();
           bw.write(b2.getLabel());
           bw.newLine();
        }
        if(!(b3.getLabel().equals("")))
        { 
           bw.write("3");
           bw.newLine();
           bw.write(b3.getLabel());
           bw.newLine();
        }
        if(!(b4.getLabel().equals("")))
        { 
           bw.write("4");
           bw.newLine();
           bw.write(b4.getLabel());
           bw.newLine();
        }
        if(!(b5.getLabel().equals("")))
        { 
           bw.write("5");
           bw.newLine();
           bw.write(b5.getLabel());
           bw.newLine();
        }
        if(!(b6.getLabel().equals("")))
        { 
           bw.write("6");
           bw.newLine();
           bw.write(b6.getLabel());
           bw.newLine();
        }
        if(!(b7.getLabel().equals("")))
        { 
           bw.write("7");
           bw.newLine();
           bw.write(b7.getLabel());
           bw.newLine();
        }
        if(!(b8.getLabel().equals("")))
        { 
           bw.write("8");
           bw.newLine();
           bw.write(b8.getLabel());
           bw.newLine();
        }
        if(!(b9.getLabel().equals("")))
        { 
           bw.write("9");
           bw.newLine();
           bw.write(b9.getLabel());
           bw.newLine();
        }
        bw.close();
        }catch(Exception ee){}
     }
     
          
    if(e.getSource()==bb2)
    {
       demo dd=new demo(a);
       setVisible(false);
       d.setVisible(false);
       try{bw2.close();}catch(Exception ee){}
       
    }
    else if(((b1.getLabel().equals(b2.getLabel())&&b2.getLabel().equals(b3.getLabel()))||(b1.getLabel().equals(b4.getLabel())&&b4.getLabel().equals(b7.getLabel())))&&((!b1.getLabel().equals(""))))
    {
         
          bb2=new Button("OK");
          bb2.setSize(50,20);
          bb2.setLocation(175,150);
          bb2.setFont(new Font("Berlin Sans FB",Font.BOLD,15));
          bb2.addActionListener(this);
          d=new Dialog(this,"winner");
          d.setLayout(null);
           d.setBackground(new Color(66,43,15));
           d.setUndecorated(true);
         d.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
           d.setSize(400,200);
           d.setLocation(400,300);
           d.add(bb2);
           d.setVisible(false);
          if(b1.getLabel().equals("X"))
          {
           l=new Label("Player 1 wins");
           try{bw2.write("1");
           bw2.newLine();}catch(Exception ee){}
           l.setSize(180,30);
           l.setLocation(110,80);
           l.setForeground(Color.WHITE);
           d.add(l);
           d.setVisible(true);
          }
          if(b1.getLabel().equals("O"))
          {
           l=new Label("Player 2 wins");
           try{bw2.write("2");
           bw2.newLine();}catch(Exception ee){}
           l.setSize(180,30);
           l.setLocation(110,80);
           l.setForeground(Color.WHITE);
           d.add(l);
           d.setVisible(true);
          }
      }
      else if(((b9.getLabel().equals(b8.getLabel())&&b8.getLabel().equals(b7.getLabel()))||(b9.getLabel().equals(b6.getLabel())&&b6.getLabel().equals(b3.getLabel())))&&((!b9.getLabel().equals(""))))
    {
         
          bb2=new Button("OK");
          bb2.setSize(50,20);
          bb2.setLocation(175,150);
          bb2.setFont(new Font("Berlin Sans FB",Font.BOLD,15));
          bb2.addActionListener(this);
          d=new Dialog(this,"winner");
          d.setLayout(null);
           d.setBackground(new Color(66,43,15));
           d.setUndecorated(true);
         d.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
           d.setSize(400,200);
           d.setLocation(400,300);
           d.add(bb2);
           d.setVisible(false);
          if(b9.getLabel().equals("X"))
          {
           l=new Label("Player 1 wins");
           try{bw2.write("1");
           bw2.newLine();}catch(Exception ee){}
           l.setSize(180,30);
           l.setLocation(110,80);
           l.setForeground(Color.WHITE);
           d.add(l);
           d.setVisible(true);
          }
          if(b9.getLabel().equals("O"))
          {
           l=new Label("Player 2 wins");
           try{bw2.write("2");
           bw2.newLine();}catch(Exception ee){}
           l.setSize(180,30);
           l.setLocation(110,80);
           l.setForeground(Color.WHITE);
           d.add(l);
           d.setVisible(true);
          }
      }
      else if(((b2.getLabel().equals(b5.getLabel())&&b5.getLabel().equals(b8.getLabel()))||(b4.getLabel().equals(b5.getLabel())&&b5.getLabel().equals(b6.getLabel()))||(b1.getLabel().equals(b5.getLabel())&&b5.getLabel().equals(b9.getLabel()))||(b3.getLabel().equals(b5.getLabel())&&b5.getLabel().equals(b7.getLabel())))&&((!b5.getLabel().equals(""))))
    {
         
          bb2=new Button("OK");
          bb2.setSize(50,20);
          bb2.setLocation(175,150);
          bb2.setFont(new Font("Berlin Sans FB",Font.BOLD,15));
          bb2.addActionListener(this);
          d=new Dialog(this,"winner");
          d.setLayout(null);
           d.setBackground(new Color(66,43,15));
           d.setUndecorated(true);
         d.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
           d.setSize(400,200);
           d.setLocation(400,300);
           d.add(bb2);
           d.setVisible(false);
          if(b5.getLabel().equals("X"))
          {
           l=new Label("Player 1 wins");
           try{bw2.write("1");
           bw2.newLine();}catch(Exception ee){}
           l.setSize(180,30);
           l.setLocation(110,80);
           l.setForeground(Color.WHITE);
           d.add(l);
           d.setVisible(true);
          }
          if(b5.getLabel().equals("O"))
          {
           l=new Label("Player 2 wins");
           try{bw2.write("2");
           bw2.newLine();}catch(Exception ee){}
           l.setSize(180,30);
           l.setLocation(110,80);
           l.setForeground(Color.WHITE);
           d.add(l);
           d.setVisible(true);
          }
      }
      else if(i>9)
      {
          bb2=new Button("OK");
          bb2.setSize(50,20);
          bb2.setLocation(175,150);
          bb2.setFont(new Font("Berlin Sans FB",Font.BOLD,15));
          bb2.addActionListener(this);
          d=new Dialog(this,"winner");
          d.setLayout(null);
           d.setBackground(new Color(66,43,15));
           d.setUndecorated(true);
         d.setFont(new Font("Berlin Sans FB",Font.BOLD,25));
           d.setSize(400,200);
           d.setLocation(400,300);
           d.add(bb2);
           d.setVisible(false);
           l=new Label("It is a draw!!!");
           try{bw2.write("0");
           bw2.newLine();}catch(Exception ee){}
           l.setSize(180,30);
           l.setLocation(110,80);
           l.setForeground(Color.WHITE);
           d.add(l);
           d.setVisible(true);
      }
      
      
  }
   public void paint(Graphics g)
  {
     g.drawImage(a,0,0,null);
     g.fillRect(0,70,400,490);
     g.setFont(new Font("Berlin Sans FB",Font.BOLD,22));
     g.setColor(Color.WHITE);
     g.drawString("PLAYER 1-X",10,30);
     g.drawString("PLAYER 2-O",250,30);
  }
  public static void main(String args[])
  {
Image rimg = null,ig = null;
     try{ 
     ig=ImageIO.read(new File("backWood.jpg"));
     rimg=ig.getScaledInstance((int)(168*2.4),(int)(300*2),Image.SCALE_FAST);
    }catch(Exception e){}
      game g=new game(rimg,false);
  }
}